src/lib/HST/CoreAlgorithm.hs

```diff
@@ -306,30 +307,29 @@ computeAlt _ _ _ []
 decomposeConPat :: Members '[Fresh, Report] r
                 => S.Pat a
                 -> Sem r (S.Pat a, [S.Pat a], [S.Pat a])
-decomposeConPat (S.PApp _ qname ps)         = do
+decomposeConPat (S.PApp s qname ps)         = do
   nvars <- replicateM (length ps) (freshVarPat genericFreshPrefix)
-  return (S.PApp S.NoSrcSpan qname nvars, nvars, ps)
-decomposeConPat (S.PInfixApp _ p1 qname p2) = failToReport $ do
+  return (S.PApp s qname nvars, nvars, ps)
+decomposeConPat (S.PInfixApp s p1 qname p2) = failToReport $ do
   nvars@[nv1, nv2] <- replicateM 2 (freshVarPat genericFreshPrefix)
   let ps = [p1, p2]
-  return (S.PInfixApp S.NoSrcSpan nv1 qname nv2, nvars, ps)
+  return (S.PInfixApp s nv1 qname nv2, nvars, ps)
 -- Decompose patterns with special syntax.
-decomposeConPat (S.PList _ ps)
-  | null ps = return (S.PList S.NoSrcSpan [], [], [])
+decomposeConPat (S.PList s ps)
+  | null ps = return (S.PList s [], [], [])
   | otherwise = do
     let (n : nv) = ps
+        nspan    = S.getSrcSpan n
         listCon  = S.Special S.NoSrcSpan $ S.ConsCon S.NoSrcSpan
-    decomposeConPat (S.PInfixApp S.NoSrcSpan n listCon (S.PList S.NoSrcSpan nv))
-decomposeConPat (S.PTuple _ bxd ps)         = do
+    decomposeConPat (S.PInfixApp nspan n listCon (S.PList S.NoSrcSpan nv))
```

Comment:
I think we should use the source span of the entire list and not the one for the first element here. We don't have a good source span for the remaining list, so we should use the source span of the entire list again in my opinion.
```suggestion
    decomposeConPat (S.PInfixApp s n listCon (S.PList s nv))
```